#! /bin/sh /usr/share/dpatch/dpatch-run
## 200_elflink_%B_fixes.dpatch by  <kirr@landau.phys.spbu.ru>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: Fixes several msgs; needed for later elflink patches
## DP: Upstream status: submitted upstream for binutils-2_20-branch

2010-08-07  Kirill Smelkov  <kirr@landau.phys.spbu.ru>

	Backport from mainline:
	2009-10-12  Roland McGrath  <roland@frob.com>

	* elflink.c (elf_link_add_object_symbols, _bfd_elf_merge_symbol):
	Fix %s that should be %B in several message formats.

@DPATCH@
diff --git a/bfd/elflink.c b/bfd/elflink.c
index c42c6e1..4a348de 100644
--- a/bfd/elflink.c
+++ b/bfd/elflink.c
@@ -1110,19 +1110,19 @@ _bfd_elf_merge_symbol (bfd *abfd,
 
       if (tdef && ntdef)
 	(*_bfd_error_handler)
-	  (_("%s: TLS definition in %B section %A mismatches non-TLS definition in %B section %A"),
+	  (_("%B: TLS definition in %B section %A mismatches non-TLS definition in %B section %A"),
 	   tbfd, tsec, ntbfd, ntsec, h->root.root.string);
       else if (!tdef && !ntdef)
 	(*_bfd_error_handler)
-	  (_("%s: TLS reference in %B mismatches non-TLS reference in %B"),
+	  (_("%B: TLS reference in %B mismatches non-TLS reference in %B"),
 	   tbfd, ntbfd, h->root.root.string);
       else if (tdef)
 	(*_bfd_error_handler)
-	  (_("%s: TLS definition in %B section %A mismatches non-TLS reference in %B"),
+	  (_("%B: TLS definition in %B section %A mismatches non-TLS reference in %B"),
 	   tbfd, tsec, ntbfd, h->root.root.string);
       else
 	(*_bfd_error_handler)
-	  (_("%s: TLS reference in %B mismatches non-TLS definition in %B section %A"),
+	  (_("%B: TLS reference in %B mismatches non-TLS definition in %B section %A"),
 	   tbfd, ntbfd, ntsec, h->root.root.string);
 
       bfd_set_error (bfd_error_bad_value);
@@ -4437,7 +4437,7 @@ error_free_dyn:
 	      if ((elf_dyn_lib_class (abfd) & DYN_NO_NEEDED) != 0)
 		{
 		  (*_bfd_error_handler)
-		    (_("%s: invalid DSO for symbol `%s' definition"),
+		    (_("%B: invalid DSO for symbol `%s' definition"),
 		     abfd, name);
 		  bfd_set_error (bfd_error_bad_value);
 		  goto error_free_vers;
@@ -12495,7 +12495,7 @@ _bfd_elf_get_dynamic_reloc_section (bfd *       abfd,
    section does not exist it is created and attached to the DYNOBJ
    bfd and stored in the SRELOC field of SEC's elf_section_data
    structure.
-   
+
    ALIGNMENT is the alignment for the newly created section and
    IS_RELA defines whether the name should be .rela.<SEC's name>
    or .rel.<SEC's name>.  The section name is looked up in the
-- 
1.7.2.1.44.g721e7

